linker/shared/src/main/scala/org/scalajs/linker/checker/IRChecker.scala

```diff
@@ -471,17 +474,25 @@ private final class IRChecker(unit: LinkingUnit, reporter: ErrorReporter) {
             DoubleType
           case String_charAt =>
             StringType
+          case Class_isInstance | Class_isAssignableFrom | Class_cast |
+              Class_newArray =>
+            ClassType(ClassClass, nullable = false)
         }
         val expectedRhsType = (op: @switch) match {
-          case Long_<< | Long_>>> | Long_>> | String_charAt => IntType
-          case _                                            => expectedLhsType
+          case Long_<< | Long_>>> | Long_>> | String_charAt | Class_newArray =>
+            IntType
+          case Class_isInstance | Class_cast =>
+            AnyType
+          case Class_isAssignableFrom =>
+            ClassType(ClassClass, nullable = true)
```

Comment:
Now that we have the ability to insert a `CheckNotNull`, should we also require a non-nullable `jl.Class` as the *rhs* of `isAssignableFrom`?

It's not as critical as the lhs, because the NPE naturally comes after evaluating both arguments, which is not true for the lhs. But it would probably be more consistent.

WDYT?

Reply:
This is quite a bit better with that change, IMO, so I did it.